resources/startup.sh

```diff
@@ -148,29 +145,41 @@ function startNexus() {
   NEXUS_PID=$!
 }
 
-function doHealthCheck() {
-  echo "wait until nexus passes all health checks"
-
-  export HTTP_BASIC_AUTH_USERNAME=$1
-  export HTTP_BASIC_AUTH_PASSWORD=$2
-
-  if ! doguctl wait-for-http --timeout 300 --method GET http://localhost:8081/nexus/service/metrics/healthcheck; then
-    echo "timeout reached while waiting for nexus to get healthy"
-    HEALTH_INFORMATION=$(curl -s -u "${HTTP_BASIC_AUTH_USERNAME}":"${HTTP_BASIC_AUTH_PASSWORD}" http://localhost:8081/nexus/service/metrics/healthcheck)
-    echo "Nexus Health information: ${HEALTH_INFORMATION}"
-    exit 1
+function waitForHealthEndpoint() {
+  local username=$1
+  local password=$2
+  local attempt_counter=0
+  local max_attempts=300
+
+  echo "Waiting until Nexus health endpoint is available (max. ${max_attempts} seconds)..."
+
+  until curl --user "${username}":"${password}" --silent --output /dev/null http://localhost:8081/nexus/service/metrics/healthcheck; do
+    if [ ${attempt_counter} -eq ${max_attempts} ];then
+      echo "Max attempts reached; exiting..."
+      exit 1
+    fi
+    attempt_counter=$((attempt_counter+1))
+    sleep 1
+  done
+
+  local health_endpoint_response
+  local unhealthy_checks
+  health_endpoint_response=$(curl --user "${username}":"${password}" --silent http://localhost:8081/nexus/service/metrics/healthcheck)
+  unhealthy_checks=$(echo "${health_endpoint_response}" | jq -c 'to_entries[] | select(.value.healthy==false) | [.key, .value.message]')
+  if [[ ${unhealthy_checks} != "" ]]; then
+    echo "WARNING! Some of the Sonatype Nexus health checks have failed:"
+    echo "${unhealthy_checks}"
   else
-    HEALTH_INFORMATION=$(curl -s -u "${HTTP_BASIC_AUTH_USERNAME}":"${HTTP_BASIC_AUTH_PASSWORD}" http://localhost:8081/nexus/service/metrics/healthcheck)
-    echo "Nexus is healthy: ${HEALTH_INFORMATION}"
+    echo "All Sonatype Nexus health checks have succeeded"
   fi
 }
 
-function waitForHealthCheck() {
-  doHealthCheck "$1" "$(<${NEXUS_DATA_DIR}/admin.password)"
+function waitForHealthEndpointAtFirstStart() {
+  waitForHealthEndpoint "$1" "$(<${NEXUS_DATA_DIR}/admin.password)"
 }
 
-function waitForHealthCheckAtSubsequentStart() {
-  doHealthCheck "$1" "$(doguctl config -e admin_password)"
+function waitForHealthEndpointAtSubsequentStart() {
+  waitForHealthEndpoint "$1" "$(doguctl config -e admin_password)"
```

Comment:
Code duplication:

The function `exportNexusPasswordFromEtcd` does the same and already exists.

Reply:
This should be resolved with a different issue because the admin password has a larger implication. The workflow of admin passwords is very likely to be change.